### ISSUE TITLE
'Subscription' is a reserved type name in 4.29.6

### ISSUE THREAD

**Issue by julien-tamade** [aws-amplify/amplify-cli#5554]
**Describe the bug**
I have a schema I've been using for quite some time now. previously i was on version 4.29.3 and encountered no issues. I updated to 4.29.6 and now I can't push anymore due to this issue: "'Subscription' is a reserved type name and currently in use within the default schema element."
My schema has a model declared as Subscription and houses all of my users' subscription info. I don't know why suddenly Subscription is a reserved type name, but changing this in my schema would be a fairly large undertaking at the moment and would potentially break my app. 

**Amplify CLI Version**
4.29.6

**To Reproduce**
create a schema using a model called Subscription in an earlier version. Upgrade to 4.29.6. try to push

**Expected behavior**
I would expect the push to happen without issue or for Amplify to provide some way to change model names easily in situations like this. 
**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version. 12.16.1

**Comment by mbecker** [comment]
Same for me!

**Comment by julien-tamade** [comment]
@mbecker did you figure out a solution by any chance? I've just been holding off on updating for now

**Comment by realityfilter** [comment]
I noticed that the code generation does strange things if you happen to have a type with name "subscription".

It contains a onUpdateTypeName, onCreateTypeName and onDeleteTypeName for all types in generated queries.js as fields for the type "subscription". 

I think this is related. Should I file another issue or is this already already tracked? My research shows nothing regarding this.

**Comment by akshbhu** [comment]
Hi @realityfilter 

You can open GH issue for your problem. I am closing this issue as it seems to be solved.

**Comment by edwardfoyle** [comment]
@julien-tamade We have released a feature flag for this which we have documented [here](https://docs.amplify.aws/cli/reference/feature-flags#validateTypeNameReservedWords). It is on by default for new and existing projects, but you can manually set it to false to unblock yourselves from using newer versions of the CLI.
However, you should still migrate away from using these reserved words in your model names as they may cause other downstream GraphQL issues.

To migrate you'll need to
1. Create a new model that will replace the model with the reserved word name
  a. Be sure to include all the same @connection @key @auth etc. directives that the old model has
2. Update the mutation resolvers for any models that have a connection to the old model to copy the connection id to the new connection field (this ensures that new connected entities will have a connection to the new model)
3. Backfill models that have a connection to the old model to have a connection to the new model as well (this ensures that existing connected entities will have a connection to the new model)
4. Create a lambda that triggers on mutations to the old model table and copies the event to the new model table (this ensures that new entities will be present in the new table). See: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.Lambda.html
5. Backfill data from the old model table into the new model table, changing __typename accordingly (this ensures that existing entities will be present in the new table)
6. Update your client code to query / mutate the new model instead of the old one
7. Remove the old model from the schema
8. Migration cleanup
  a. Remove the old connection fields from the tables of models that had a connection to the old model
  b. Remove the lambda trigger

When going from 6 to 7, if you have an iOS or Android client, you'll need to make sure that the old model is not in use anymore before removing it.

**Comment by julien-tamade** [comment]
Hi @edwardfoyle I'm just getting around to making this migration. Can you provide tips for how to do section 2?

"Update the mutation resolvers for any models that have a connection to the old model to copy the connection id to the new connection field (this ensures that new connected entities will have a connection to the new model)"

Does this involve manually changing the autogenerated resolvers?
